VotingApplication/VotingApplication.Web/Scripts/Controllers/AddVoterOptionDialogController.js

```diff
@@ -32,7 +32,8 @@
 
             $scope.options.push(newVoterOption);
 
-            VoteService.addVoterOption($scope.ngDialogData.pollId, newVoterOption);
+            VoteService.addVoterOption($scope.ngDialogData.pollId, newVoterOption)
+                .then($scope.notifyOptionAdded());
```

Comment:
The chaining might be causing an issue here;

It looks like the then() gets called before the option has been added to the database, so it is always one out